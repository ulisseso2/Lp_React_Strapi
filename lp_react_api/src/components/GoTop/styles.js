import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: red;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background-color: red;
  `}
`;
