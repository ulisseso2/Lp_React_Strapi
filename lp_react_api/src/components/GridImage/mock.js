export default {
  title: 'Grid de Imagens',
  description: `Lorem ipsum dolor sit, amet consectetur adipisicing elit. Adipisci numquam explicabo error omnis cumque.`,
  image: [
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=1',
      alt: 'Foto 1',
    },
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=2',
      alt: 'Foto 2',
    },
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=3',
      alt: 'Foto 3',
    },
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=4',
      alt: 'Foto 3',
    },
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=5',
      alt: 'Foto 3',
    },
    {
      srcImg: 'https://source.unsplash.com/random/800x800?r=6',
      alt: 'Foto 3',
    },
  ],
};
