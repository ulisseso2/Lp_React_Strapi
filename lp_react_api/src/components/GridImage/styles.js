import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css``}
`;

export const Title = styled.div`
  ${({ theme }) => css``}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    align-items: center;
  `}
`;
export const GridImg = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    margin: 10px;
  `}
`;

export const Img = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
