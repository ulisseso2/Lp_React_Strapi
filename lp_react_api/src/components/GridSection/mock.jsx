export default {
  background: false,
  title: 'My grid',
  description: `Assumenda, animi praesentium voluptas velit voluptatum quam repudiandae, impedit minus consequatur iusto atque voluptatem numquam, aliquam tempore ratione culpa facere asperiores inventore?`,
  grid: [
    {
      title: 'Teste a',
      description: `Lorem 1 ipsum dolor sit amet consectetur adipisicing elit. Architecto rem cupiditate dolorum voluptatem in pariatur cumque exercitationem molestias aliquid. Magni veniam numquam, reiciendis voluptas aliquid ut saepe vero vel ratione!`,
    },
    {
      title: 'Teste b',
      description: `Lorem 2 ipsum dolor sit amet consectetur adipisicing elit. Architecto rem cupiditate dolorum voluptatem in pariatur cumque exercitationem molestias aliquid. Magni veniam numquam, reiciendis voluptas aliquid ut saepe vero vel ratione!`,
    },
    {
      title: 'Teste c',
      description: `Lorem 3 ipsum dolor sit amet consectetur adipisicing elit. Architecto rem cupiditate dolorum voluptatem in pariatur cumque exercitationem molestias aliquid. Magni veniam numquam, reiciendis voluptas aliquid ut saepe vero vel ratione!`,
    },
  ],
};
