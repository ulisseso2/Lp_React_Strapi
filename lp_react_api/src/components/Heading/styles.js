import styled from 'styled-components';
import { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, colorDark }) => css`
    color: ${colorDark ? theme.colors.hot : theme.colors.dark};
  `};
`;
// styled component do componente Heading
