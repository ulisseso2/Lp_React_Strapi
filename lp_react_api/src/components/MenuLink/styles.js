import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.small};
    text-decoration: none;
    padding: ${theme.spacings.small};
    color: ${theme.colors.promaryColor};
    font-weight: 400;
    position: relative;
    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 50%;
      height: 0.2rem;
      width: 0px;
      background-color: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
