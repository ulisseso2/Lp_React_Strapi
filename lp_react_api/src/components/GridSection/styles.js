import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from '../Heading/styles';
import { Container as Textcomponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Textcomponent} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridElement = styled.div`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 6rem;
      font-weight: 900;
      left: -4rem;
      top: -1.5rem;
      transform: rotate(7deg);
    }

`;
