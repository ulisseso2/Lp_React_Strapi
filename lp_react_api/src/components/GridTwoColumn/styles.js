import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.font.media.ltmedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
      color: ${background ? theme.colors.white : theme.colors.primaryColor};
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css``}
`;
export const ImgContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    max-width: 500px;
  `}
`;
export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 500px;
  `}
`;
