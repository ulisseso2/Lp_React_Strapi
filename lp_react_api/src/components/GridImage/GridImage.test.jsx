import { screen } from '@testing-library/react';
import { GridImage } from '.';
import { renderTheme } from '../../styles/render.theme';

import mock from './mock';
import { GridImg } from './styles';

describe('<GridImage>', () => {
  it('should render with default values', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });
  it('should render with default values', () => {
    const { container } = renderTheme(
      <GridImg {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
