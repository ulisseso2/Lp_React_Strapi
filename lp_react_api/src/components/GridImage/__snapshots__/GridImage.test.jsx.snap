// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage> should render with default values 1`] = `
<div>
  <div
    class="sc-gKXOVf dgAgri"
    id=""
  >
    <div
      class="sc-jSMfEi cRkChI"
    >
      <div
        class="sc-bczRLJ edGOPd"
      >
        <div
          class="sc-gsnTZi gaQnzH"
        >
          <h2
            class="sc-iBkjds xsJLf"
          >
            Grid de Imagens
          </h2>
          <div
            class="sc-ftvSup fPvrFX"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Adipisci numquam explicabo error omnis cumque.
          </div>
        </div>
        <div
          class="sc-dkzDqf jjpYOo"
        >
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 1"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 2"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 3"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 3"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 3"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="sc-hKMtZM gtoZYg"
          >
            <img
              alt="Foto 3"
              class="sc-eCYdqJ likfFv"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`<GridImage> should render with default values 2`] = `
<div>
  <div
    class="sc-hKMtZM gtoZYg"
    title="Grid de Imagens"
  />
   
</div>
`;
